http-netty/src/main/java/io/micronaut/http/netty/NettyMutableHttpResponse.java

```diff
@@ -287,6 +301,18 @@ public <T> MutableHttpResponse<T> body(@Nullable T body) {
         return (MutableHttpResponse<T>) this;
     }
 
+    @Override
+    public MutableHttpResponse<B> contentType(MediaType mediaType) {
+        if (mediaType == null) {
+            headers.remove(HttpHeaderNames.CONTENT_TYPE);
+        } else {
+            // optimization for content type validation
+            mediaType.validate(() -> NettyHttpHeaders.validateHeader(HttpHeaderNames.CONTENT_TYPE, mediaType));
+            headers.setUnsafe(HttpHeaderNames.CONTENT_TYPE, mediaType);
```

Comment:
Why do this validation here? Why not put this logic in `NettyHttpHeaders` so that there exists no API on `NettyHttpHeaders` that allows for an unsafe use?

Reply:
these are internal non-public APIs inaccessible to users, this PR was partly to optimise performance, so excessive validation when it is not needed is not great for performance.